tests/conftest.py

```diff
@@ -59,3 +65,48 @@ def pytest_collection_modifyitems(config, items):
         for item in items:
             if "slow" in item.keywords:
                 item.add_marker(skip_slow)
+
+
+def num_nonsample_muts(ts):
+    return np.sum(np.logical_not(np.isin(ts.tables.mutations.node, ts.samples())))
+
+
+@fixture(scope="session")
+def small_ts_fixture():
+    """
+    A simple 1-tree sequence with at least 2 inference sites
+    (i.e. mutations above a non-sample node)
+    """
+    ts = msprime.simulate(10, mutation_rate=1, random_seed=1)
+    assert num_nonsample_muts(ts) > 1
+    return ts
+
+
+@fixture(scope="session")
```

Comment:
Is `SampleData` immutable? Otherwise you can't set `scope` to session.

Reply:
Thanks Ben. You can't modify a SampleData file - you have to take a copy. There's a flag ".READ_MODE" on it that is set once it is made, and all write methods check this before doing anything.  ".READ_MODE" can't be unset AFAIK. So I think the answer is that practically, it is immutable, although perhaps that's not known by Python.